### ISSUE TITLE
Zookeeper systemd start file is not dependent on networking

### ISSUE THREAD

**Issue by eyal-lupu** [deric/puppet-zookeeper#47]
hi

We have seen an issue (CentOS 7.1) in which ZK fails to start since the unit file is not dependent on the networking being available (that is 'network-online.target'). Because of that the process starts before the bond interfaces do and fails.

If this is agreed as a bug I am happy to provide a pull request with a fix.

**Comment by deric** [comment]
Seems more like a CentOS bug to me. We can provide a way how to override system defaults. It would be good, such behaviour  could be turned off.

**Comment by eyal-lupu** [comment]
Hi
I am not sure this is a CentoOS bug. A quick search on both CentOS and Fedora shows this is a general practice for system services, to name a few:

- dnsmasq
- sshd
- ntpdate
- nslcd
- etc....

However it is worth noticing that all of the above are dependent on network.target and not on network-online.target as my zookeeper instance requires. The reason for that is the duality of some of the newer Linux systems. E.g in my case some of the networking (the teaming interfaces) are started by /etc/rc.d files which happens after systemd units. 

IMO this should behave as:

By default the unit file should include (as compatible with all system services):
````
[Unit]
After=network.target
````

This will probably work for most cases. However I would also suggest adding a class parameter to indicate if network-online.target is required, if so include the following in the file:

````
[Unit]
Wants=network-online.target
After=network-online.target
````

I could provide a pull request for that if we decide to go that way.

**Comment by deric** [comment]
Sounds good to me. Overriding configuration in `/etc/systemd/system/...` is definitely preferred way. Such patch might be compatible with other distributions using systemd. Btw. do you use standard interface bonding (from linux kernel)?

**Comment by deric** [comment]
Closed by #48.
